src/ast/types/blocks-and-statements.ts

```diff
@@ -1,39 +1,86 @@
 import { Identifier, UnannType } from "./classes";
 
-export type BlockStatement = LocalVariableDeclarationStatement;
+export interface Block {
+  kind: "Block";
+  blockStatements: Array<BlockStatement>;
+}
+export type BlockStatement = LocalVariableDeclarationStatement | Statement;
+
 export interface LocalVariableDeclarationStatement {
   kind: "LocalVariableDeclarationStatement";
   localVariableType: LocalVariableType;
-  variableDeclarationList: VariableDeclarator;
+  variableDeclaratorList: Array<VariableDeclarator>;
+}
+
+export type Statement = StatementWithoutTrailingSubstatement | IfStatement | WhileStatement | DoStatement | ForStatement;
+
+export interface IfStatement {
+  kind: "IfStatement";
+  condition: Expression;
+  consequent: Statement;
+  alternate: Statement;
+}
+
+export interface WhileStatement {
+  kind: "WhileStatement";
+  condition: Expression;
+  body: Statement;
+}
+
+export interface DoStatement {
+  kind: "DoStatement";
+  condition: Expression;
+  body: Statement;
+}
+
+export interface ForStatement {
+  kind: "ForStatement";
+}
+
+export type StatementWithoutTrailingSubstatement = ExpressionStatement;
+
+export type ExpressionStatement = MethodInvocation;
+
+export interface MethodInvocation {
+  kind: "MethodInvocation";
+  identifier: Identifier
+  argumentList: ArgumentList;
 }
 
+export type ArgumentList = Array<Expression>;
+
 export type LocalVariableType = UnannType;
 
 export interface VariableDeclarator {
+  kind: "VariableDeclarator";
   variableDeclaratorId: VariableDeclaratorId;
   variableInitializer: VariableInitializer;
 }
+
 export type VariableDeclaratorId = Identifier;
 export type VariableInitializer = Expression;
-export type Expression = Literal | BinaryExpression | UnaryExpression;
+
+export type Expression = Primary | BinaryExpression | UnaryExpression;
```

Comment:
There is another `Expression` defined in `ast.ts`, perhaps can remove that?

Reply:
Sure, removed.